autogen/agentchat/chat.py

```diff
@@ -180,9 +183,18 @@ def initiate_chats(chat_queue: List[Dict[str, Any]]) -> List[ChatResult]:
     while current_chat_queue:
         chat_info = current_chat_queue.pop(0)
         _chat_carryover = chat_info.get("carryover", [])
+        carryover_indexes = chat_info.get("carryover_indexes", [])
         if isinstance(_chat_carryover, str):
             _chat_carryover = [_chat_carryover]
-        chat_info["carryover"] = _chat_carryover + [r.summary for r in finished_chats]
+        chat_info["carryover"] = _chat_carryover
+        if chat_info.get("finished_chat_summary", True):
+            if not carryover_indexes:
+                chat_info["carryover"] = chat_info["carryover"] + [r.summary for r in finished_chats]
+            else:
+                for index in carryover_indexes:
+                    if not (0 <= index < len(finished_chats)):
+                        raise IndexError(f"Invalid carryover_index: {index}. Index out of range for finished_chats")
```

Comment:
The design of `carryover_indexes` seems not very intuitive and may cause problems. This index error problem is one. The `finished_chats` is typically empty when people use `initiate_chats`. When `finished_chats` is empty, this IndexError is guaranteed to be triggered.

Reply:
with  finished_chat_indexes_to_exclude_from_carryover this issue will also get resolve.